service/src/main/java/com/commercetools/pspadapter/payone/notification/common/CaptureNotificationProcessor.java

```diff
@@ -28,37 +30,64 @@ public CaptureNotificationProcessor(TenantFactory tenantFactory, TenantConfig te
         super(tenantFactory, tenantConfig, transactionStateResolver);
     }
 
-        @Override
+    @Override
     protected boolean canProcess(final Notification notification) {
         return NotificationAction.CAPTURE.equals(notification.getTxaction());
     }
 
     @Override
     protected ImmutableList<UpdateAction<Payment>> createPaymentUpdates(final Payment payment,
                                                                         final Notification notification) {
-        final ImmutableList.Builder<UpdateAction<Payment>> listBuilder = ImmutableList.builder();
-        listBuilder.addAll(super.createPaymentUpdates(payment, notification));
+        final ImmutableList.Builder<UpdateAction<Payment>> actionsBuilder = ImmutableList.builder();
+        actionsBuilder.addAll(super.createPaymentUpdates(payment, notification));
 
         final List<Transaction> transactions = payment.getTransactions();
         final String sequenceNumber = toSequenceNumber(notification.getSequencenumber());
 
         return findMatchingTransaction(transactions, TransactionType.CHARGE, sequenceNumber)
-                .map(transaction -> listBuilder.build())
-                .orElseGet(() -> {
-                    final MonetaryAmount amount = MoneyImpl.of(notification.getPrice(), notification.getCurrency());
+                .map(transaction -> updateChargeTransactionState(transaction, notification, actionsBuilder))
+                .orElseGet(() -> createChargeTransaction(notification, actionsBuilder, sequenceNumber))
+                .build();
+    }
+
+    /**
+     * Add {@link ChangeTransactionState} action to {@code actionsBuilder} if current transaction is still in different
+     * state.
```

Comment:
what do you mean "still in different state"? : 
1.  different state than what? 
2. "Still"? Does it change at a certain point? what point?

Reply:
1. a5310c21

2. The whole class is a _NotificationProcessor_, "still" and "certain point" mean until notification arrived.